File: Python/khudi/color/RGBColor.java
package khudi.color;

import java.lang.Math;

/**
 * @section DESCRIPTION
 *
 * The RGBColor Class.
 * Provides the RGBColor class ....
 *
 */
public class RGBColor
{
	public double red, green, blue;

	public RGBColor()
	{
		red = green = blue = 0.0;
	}

	public RGBColor(double r, double g, double b)
	{
		red = r;
		green = g;
		blue = b;
	}

	public RGBColor(RGBColor rgb)
	{
		red = rgb.red;
		green =rgb.green ;
		blue = rgb.blue;
	}

	public RGBColor plus (double d)
	{
		return ( new RGBColor(red + d, green + d, blue + d) );
	}

	public RGBColor plus (RGBColor rgb)
	{
		return ( new RGBColor(red + rgb.red, green + rgb.green, blue + rgb.blue) );
	}

	public RGBColor minus (double d)
	{
		return ( new RGBColor(red - d, green - d, blue - d) );
	}

	public RGBColor minus (RGBColor rgb)
	{
		return ( new RGBColor(red - rgb.red, green - rgb.green, blue - rgb.blue) );
	}

	public RGBColor mul (double d)
	{
		return ( new RGBColor(red * d, green * d, blue * d) );
	}

	public RGBColor mul (RGBColor rgb)
	{
		return ( new RGBColor(red * rgb.red, green * rgb.green, blue * rgb.blue) );
	}

	public RGBColor div (double d)
	{
		return ( new RGBColor(red / d, green / d, blue / d) );
	}

	public boolean equal (RGBColor rgb)
	{
		return ( red == rgb.red && green == rgb.green && blue == rgb.blue );
	}

	public RGBColor power(double p)
	{
		return ( new RGBColor(Math.pow(red, p), Math.pow(green, p), Math.pow(blue, p)) );
	}

}

File: Python/khudi/color/RGBColor.py
#!/usr/bin/env python
""" generated source for module RGBColor """
from __future__ import print_function
# package: khudi.color
# 
#  * @section DESCRIPTION
#  *
#  * The RGBColor Class.
#  * Provides the RGBColor class ....
#  *
#  
class RGBColor(object):
    """ generated source for class RGBColor """
    red = float()
    green = float()
    blue = float()


    def __init__(self, r=0.0, g=0.0, b=0.0):
        """ generated source for method __init__ """
        if isinstance(r, RGBColor):
            self.red = r.red
            self.green = r.green
            self.blue = r.blue
        else:
            self.red = r
            self.green = g
            self.blue = b



    def plus(self, d):
        """ generated source for method plus """
        if isinstance(d, RGBColor):
            return (RGBColor(self.red + d.red, self.green + d.green, self.blue + d.blue))

        return (RGBColor(self.red + d, self.green + d, self.blue + d))

    def minus(self, d=None):
        """ generated source for method minus """

        if isinstance(d, RGBColor):
            return (RGBColor(self.red - d.red, self.green - d.green, self.blue - d.blue))

        else:
            return (RGBColor(self.red - d, self.green - d, self.blue - d))


    def mul(self, d):
        """ generated source for method mul """
        if isinstance(d, RGBColor):
            rgb = d
            return (RGBColor(self.red * rgb.red, self.green * rgb.green, self.blue * rgb.blue))

        return (RGBColor(self.red * d, self.green * d, self.blue * d))


    def div(self, d):
        """ generated source for method div """
        return (RGBColor(self.red / d, self.green / d, self.blue / d))

    def equal(self, rgb):
        """ generated source for method equal """
        return (self.red == rgb.red and self.green == rgb.green and self.blue == rgb.blue)

    def power(self, p):
        """ generated source for method power """
        return (RGBColor(pow(self.red, p), pow(self.green, p), pow(self.blue, p)))

